File: app/src/main/java/com/constraint/vagabond/main/search/adapter/RecreationalAreaAdapter.java
package com.constraint.vagabond.main.search.adapter;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.constraint.vagabond.R;
import com.constraint.vagabond.data.entities.RecreationalAreaList;

public class RecreationalAreaAdapter extends RecyclerView.Adapter<AreaHolder> {

  private AreaAdapterPresenter areaAdapterPresenter;

  public RecreationalAreaAdapter(RecreationalAreaList recreationalAreaList) {
    this.areaAdapterPresenter = new AreaAdapterPresenter(recreationalAreaList);
  }

  @NonNull
  @Override
  public AreaHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
    return new AreaHolder(
        LayoutInflater.from(parent.getContext()).inflate(R.layout.list_view, parent, false));
  }

  @Override
  public void onBindViewHolder(@NonNull AreaHolder holder, int position) {
    areaAdapterPresenter.onBindDataToAdapter(holder, position);
  }

  @Override
  public int getItemCount() {
    return areaAdapterPresenter.getAreaCount();
  }
}

File: app/src/main/java/com/constraint/vagabond/main/search/adapter/RecreationalAreaAdapter.kt
package com.constraint.vagabond.main.search.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

import com.constraint.vagabond.R
import com.constraint.vagabond.data.entities.RecreationalAreaList

class RecreationalAreaAdapter(recreationalAreaList: RecreationalAreaList) : RecyclerView.Adapter<AreaHolder>() {

    private val areaAdapterPresenter: AreaAdapterPresenter = AreaAdapterPresenter(recreationalAreaList)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AreaHolder {
        return AreaHolder(
                LayoutInflater.from(parent.context).inflate(R.layout.list_view, parent, false))
    }

    override fun onBindViewHolder(holder: AreaHolder, position: Int) {
        areaAdapterPresenter.onBindDataToAdapter(holder, position)
    }

    override fun getItemCount(): Int {
        return areaAdapterPresenter.areaCount
    }
}
